src/calculation/ClauseResultsBuilder.ts

```diff
@@ -363,100 +372,162 @@ export function prettyResult(result: any | null, indentLevel?: number, keyIndent
+      return prettyResultReturn;
     }
-    prettyResultReturn += `${currentIndentation}}`;
-    return prettyResultReturn;
   }
   if (result) {
     return JSON.stringify(result, null, 2);
   }
   return 'null';
 }
 
+/**
+ * Uses the pieces of a code object to show code, system, and display in a pretty way
+ * for whichever properties are available.
+ */
+export function prettyCode(
+  code: string,
+  system: string | undefined,
+  display: string | undefined,
+  includeType: boolean
+) {
+  const typeStr = includeType ? 'CODE: ' : '';
+  const displayStr = display ? `, ${display}` : '';
+  let systemStr = 'UNDEFINED_SYSTEM';
+  if (system) {
+    if (system in systemMap) {
+      systemStr = systemMap[system];
+    } else {
+      systemStr = system;
+    }
+  }
+  return `${typeStr}${systemStr} ${code}${displayStr}`;
+}
+
+/**
+ * Uses the pieces of a codeable concept object to show concept display text (if available)
+ * and the contained codings that belong to the concept in a pretty way
+ */
+export function prettyConcept(
+  conceptTxt: string | undefined,
+  conceptCodings: any[] | undefined,
+  includeType: boolean,
+  indentLevel?: number,
+  keyIndent?: number
+) {
+  const keyDiff = 2; //offset between concept key and code key including bracket
+  const keyIndentation = Array((keyIndent ?? 0) + keyDiff).join(' ');
+  let prettyConcept = includeType ? 'CONCEPT: ' : '';
+  if (conceptTxt) {
+    prettyConcept = prettyConcept.concat(`${conceptTxt}`);
+  }
+  if (conceptCodings) {
+    prettyConcept = prettyConcept.concat(
+      `\n${keyIndentation}${prettyResult(conceptCodings, includeType, indentLevel, (keyIndent ?? 0) + keyDiff + 1)}` //add one to account for array bracket
+    );
+  }
+  return prettyConcept;
+}
+
+/**
+ * Function to help prettyResult handle FHIRObject(s) specifically, handling similar types, but
+ * in a FHIRObject structure rather than a CQL object structure. Uses similar inputs put expects
+ * a FHIRObject result to be passed in.
+ */
+export function prettyFHIRObject(
+  result: FHIRObject,
+  includeType = true,
+  indentLevel?: number,
+  keyIndent?: number
+): string {
+  const resourceType = result.getTypeInfo()._name;
+  const keyIndentation = Array(keyIndent).join(' ');
+  if (resourceType === 'Period') {
+    const typeStr = includeType ? 'INTERVAL: ' : '';
+    return `${typeStr}${prettyResult(result.get('start'))} - ${prettyResult(result.get('end'))}`;
+  } else if (resourceType === 'dateTime') {
+    return `${moment.utc(result.get('value').toString()).format('MM/DD/YYYY h:mm A')}`;
+  } else if (resourceType === 'Coding') {
+    return prettyCode(
+      result.get('code')?.['value'],
+      result.get('system')?.['value'],
+      result.get('display')?.['value'],
+      includeType
+    );
+  } else if (resourceType === 'CodeableConcept') {
+    return prettyConcept(result.get('text')?.['value'], result.get('coding'), includeType, indentLevel, keyIndent);
+  }
+
+  // Handle resources
```

Comment:
I think there are a lot of data types missing from this handling. Some more realistic than others, but I think the assumption that we are dealing with FHIR resources after this if/else branch is not quite right. What if we have a FHIR Duration, string, Identifier, Reference, code, id, etc.? I tested with a big example and got a couple of errors where the code assumes we are dealing with a FHIR resource type. I'll include the bundle and patient I was testing with here, I think it covers a lot more cases than the integration test

[pretty-kitchen-sink.zip](https://github.com/projecttacoma/fqm-execution/files/11885689/pretty-kitchen-sink.zip)

I don't necessarily think we need to add handling for each one of these, but I would definitely prefer if this could could ensure that `resourceType` is a valid key of `parsedCodePaths` before trying to access it so we avoid runtime errors.